pkg/controller/build/build_controller.go

```diff
@@ -0,0 +1,1332 @@
+
+}
+
+func (ctrl *Controller) updateBuild(old, cur interface{}) {
+	build := old.(*buildv1.Build)
+
+	glog.Infof("Updated a build: %s", build.Name)
+	// Builds will move through phases which cause them to change
+	// Most of those phases are standard/good, but some of them are bad
+	// We want to know if we end up in a bad phase and need to retry
+	ctrl.enqueuePoolIfBuildProblems(build)
+
+}
+
+func (ctrl *Controller) deleteBuild(obj interface{}) {
+	build := obj.(*buildv1.Build)
+
+	glog.Infof("Deleted a build: %s", build.Name)
+
+}
+
+// Buildconfigs
+
+func (ctrl *Controller) addBuildConfig(obj interface{}) {
+	buildconfig := obj.(*buildv1.BuildConfig)
+
+	glog.Infof("Added a buildconfig: %s", buildconfig.Name)
+
+}
+
+func (ctrl *Controller) updateBuildConfig(old, cur interface{}) {
+	buildconfig := old.(*buildv1.BuildConfig)
+	newbuildconfig := cur.(*buildv1.BuildConfig)
+
+	glog.Infof("Updated a buildconfig: %s", buildconfig.Name)
+
+	// Every time a buildconfig is instantiated it bumps the generation, so it always looks like it's changing
+	// For now we really only care if the user edited the dockerfile, and that string is a pointer
+	if buildconfig.Spec.Source.Dockerfile != nil && newbuildconfig.Spec.Source.Dockerfile != nil {
+		if *buildconfig.Spec.Source.Dockerfile != *newbuildconfig.Spec.Source.Dockerfile {
+
+			glog.Infof("The dockerfile for buildconfig %s changed, triggering a build", buildconfig.Name)
+			// TODO(jkyros); If this is the mco content, we need the machineconfig name
+			// so go get the image from that imagestream and put the name in. Otherwise just start it.
+
+			br := &buildv1.BuildRequest{
+				//TypeMeta:    metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{Name: buildconfig.Name},
+				//Env: []corev1.EnvVar{whichConfig},
+				TriggeredBy: []buildv1.BuildTriggerCause{
+					{Message: "The machine config controller"},
+				},
+				DockerStrategyOptions: &buildv1.DockerStrategyOptions{
+					//BuildArgs: []corev1.EnvVar{whichConfig},
+					//NoCache:   new(bool),
+				},
+			}
+
+			_, err := ctrl.buildclient.BuildV1().BuildConfigs(ctrlcommon.MCONamespace).Instantiate(context.TODO(), br.Name, br, metav1.CreateOptions{})
+			if err != nil {
+				glog.Errorf("Failed to trigger image build: %s", err)
+			}
+		}
+	}
+
+}
+
+func (ctrl *Controller) deleteBuildConfig(obj interface{}) {
+	buildconfig := obj.(*buildv1.BuildConfig)
+
+	glog.Infof("Deleted a buildconfig: %s", buildconfig.Name)
+
+}
+
+// experimentalAddEntitlements grabs the cluster entitlement certificates out of the openshift-config-managed namespace and
+// stuffs them into a machineconfig for our layered pool, so we can have entitled builds. This is a terrible practice, and
+// we should probably just sync the secrets into our namespace so our builds can use them directly rather than expose them via machineconfig.
+func (ctrl *Controller) experimentalAddEntitlements(pool *mcfgv1.MachineConfigPool) {
```

Comment:
Talking with Colin, we will need to think about the RHEL9 ramifications of this -- because it will probably be wrong :smile:

Reply:
Beyond just the RHEL8 versus RHEL9 versus e.g. C9S issues, I think we also need to handle the case of disconnected installs where a user already has a local RHEL mirror.  Us trying to overwrite `redhat.repo` would then just be broken.

I think we need to de-couple "automatically set up repos" from layering; the former needs to be a different opt-in mechanism.